lib/miq_automation_engine/service_models/miq_ae_service_vm.rb

```diff
@@ -8,5 +8,26 @@ def add_to_service(service)
     def remove_from_service
       ar_method { wrap_results(@object.direct_service.try(:remove_resource, @object)) }
     end
+
+    def create_snapshot(name, desc = nil, memory = false)
+      snapshot_operation(:create_snapshot, :name => name, :description => desc, :memory => !!memory)
+    end
+
+    def remove_all_snapshots
+      snapshot_operation(:remove_all_snapshots)
+    end
+
+    def remove_snapshot(snapshot_id)
+      snapshot_operation(:remove_snapshot, :snap_selected => snapshot_id)
+    end
+
+    def revert_to_snapshot(snapshot_id)
+      snapshot_operation(:revert_to_snapshot, :snap_selected => snapshot_id)
+    end
+
+    def snapshot_operation(task, options = {})
+      options.merge!(:ids => [id], :task => task.to_s)
```

Comment:
This does not provide any feedback to the caller if snapshots are not supported on the provider.  Since we are moving from a specific provider implementation to a generic VM one we should add a call at the top of this method to raise an error if not supported.

LIke this (also address rubocop issues):
```ruby
def snapshot_operation(task, options = {})
  raise "#{task} operation not supported for #{self.class.name}" unless @object.supports_snapshots?

  options[:ids]  = [id]
  options[:task] = task.to_s
  Vm.process_tasks(options)
end
```